### DISCUSSION TITLE
Download briefcase for editing

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#6055] [Q&A]
Using the desktop itwin app template i noticed that requesting the briefcase is done like this
```Typescript
await NativeApp.requestDownloadBriefcase(
        iTwinId,
        iModelId,
        { syncMode: SyncMode.PullOnly, fileName, progressCallback },
        IModelVersion.latest()
      );
```

So i changed `SyncMode.PullOnly` tp `SyncMode.PullAndPush`, but when downloading the briefcase i am getting this error
`Access denied due to invalid access_token. One or more required scopes are missing. Make sure to provide a valid token for this API endpoint`.

Here are my scopes : `itwins:read projects:read itwins:modify projects:modify users:read sensor-data:read sensor-data:modify savedviews:modify imodels:modify storage:modify storage:read savedviews:read library:modify realitydata:read library:read imodels:read realitydata:modify itwinjs imodelaccess:read`.

**Comment by calebmshafer** [comment]
@salah-walid can you provide the x-correlation-id of the request being sent for download? The list of scopes seems correct as you need both `imodels:read imodels:modify`.

**Reply by salah-walid** [reply]
I can't find any network activity when downloading the briefcase, where can i find this `x-correlation-id` ?

Also to be noted i am generating the app with this command `npx create-react-app app --template @itwin/desktop-viewer --scripts-version @bentley/react-scripts`, and changing the ITWIN_VIEWER_SCOPE env variable for the ones mentioned above.

**Reply by AnDuong249** [reply]
@salah-walid Hi, can you double check if your app permissions on developer.bentley.com have the scopes needed (`imodels:read imodels:modify`) and that it also matches the scopes in your env variable as well?
